### PR TITLE
Consider metric evaluations in query endpoints & update PromQL engine namespace

### PR DESCRIPTION
**Pull request by Harkishen-Singh**
Signed-off-by: Harkishen-Singh <harkishensingh@hotmail.com>

Fixes: #439

### CODE REVIEW

File: pkg/promql/engine.go [hunk 1]
```diff
@@ -69,7 +71,7 @@ type Querier interface {
 }
 
 const (
-	namespace            = "prometheus"
+	namespace            = util.PromNamespace
```
Comment (Harkishen-Singh) [inline pkg/promql/engine.go:74]: Breaking change but necessary. @cevian

Comment (Harkishen-Singh) [reply pkg/promql/engine.go:74]: We do not want our PromQL engine stats to have names that match exactly with that of Prometheus's PromQL engine.

Comment (cevian) [reply pkg/promql/engine.go:74]: Please make a note on the release notes draft

File: pkg/api/query.go [hunk 2]
```diff
@@ -14,12 +14,12 @@ import (
...
 
-func Query(conf *Config, queryEngine *promql.Engine, queryable promql.Queryable) http.Handler {
-	hf := corsWrapper(conf, queryHandler(queryEngine, queryable))
+func Query(conf *Config, queryEngine *promql.Engine, queryable promql.Queryable, metrics *Metrics) http.Handler {
+	hf := corsWrapper(conf, queryHandler(queryEngine, queryable, metrics))
 	return gziphandler.GzipHandler(hf)
 }
 
-func queryHandler(queryEngine *promql.Engine, queryable promql.Queryable) http.HandlerFunc {
+func queryHandler(queryEngine *promql.Engine, queryable promql.Queryable, metrics *Metrics) http.HandlerFunc {
```
Comment (cevian) [inline pkg/api/query.go:22]: This is good. But I think we need to add two metrics: the equivalent of QueryBatchDuration (call it QueryDuration) that times the queries. And the equivalent of InvalidReadReqs (call it InvalidQueryReqs) which traces the queries with incorrect parameters.

File: pkg/api/query_range.go [hunk 3]
```diff
@@ -14,12 +14,12 @@ import (
 	"github.com/timescale/promscale/pkg/promql"
 )
 
-func QueryRange(conf *Config, queryEngine *promql.Engine, queriable promql.Queryable) http.Handler {
-	hf := corsWrapper(conf, queryRange(queryEngine, queriable))
+func QueryRange(conf *Config, queryEngine *promql.Engine, queryable promql.Queryable, metrics *Metrics) http.Handler {
```
Comment (cevian) [inline pkg/api/query_range.go:17]: similar to above we should add: QueryRangeDuration and InvalidQueryRangeReqs

**Review by cevian — CHANGES_REQUESTED** [review]

File: pkg/promql/engine.go (hunk 1 above)
Comment (Harkishen-Singh) [reply pkg/promql/engine.go:74]: Updated the release. I hope the sentence is not too long.

**Review by cevian — APPROVED** [review]

**Pull request merged**